cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -0,0 +1,177 @@
+import net.corda.core.utilities.unwrap
+import java.util.*
+
+/**
+ * This flow allows two parties (the [Initiator] and the [Acceptor]) to come to an agreement about the IOU encapsulated
+ * within an [IOUState].
+ *
+ * In our simple example, the [Acceptor] always accepts a valid IOU.
+ *
+ * These flows have deliberately been implemented by using only the call() method for ease of understanding. In
+ * practice we would recommend splitting up the various stages of the flow into sub-routines.
+ *
+ * All methods called within the [FlowLogic] sub-class need to be annotated with the @Suspendable annotation.
+ */
+object PaybackFlow {
+
+    @InitiatingFlow
+    @StartableByRPC
+    class Initiator(val iouStateLinearId: String) : FlowLogic<SignedTransaction>() {
+        companion object {
+            object CREATING_COMMAND : Step("Pull data from vault and create command.")
+            object GENERATING_TRANSACTION : Step("Generating transaction based on IOU state.")
+            object VERIFYING_TRANSACTION : Step("Verifying contract constraints.")
+            object SIGNING_TRANSACTION : Step("Signing transaction with our private key.")
+            object INITIALISE_OTHER_PARTY_FLOW : Step("Send the state to the counterparty, and receive it back with their signature")
+
+            object GATHERING_SIGS : Step("Gathering the counterparty's signature.") {
+                override fun childProgressTracker() = CollectSignaturesFlow.tracker()
+            }
+
+            object PAYBACK_COMMUNICATION : Step("Sending payback value to lender.")
+
+            object FINALISING_TRANSACTION : Step("Obtaining notary signature and recording transaction.") {
+                override fun childProgressTracker() = FinalityFlow.tracker()
+            }
+
+            fun tracker() = ProgressTracker(
+                    CREATING_COMMAND,
+                    GENERATING_TRANSACTION,
+                    VERIFYING_TRANSACTION,
+                    SIGNING_TRANSACTION,
+                    INITIALISE_OTHER_PARTY_FLOW,
+                    GATHERING_SIGS,
+                    PAYBACK_COMMUNICATION,
+                    FINALISING_TRANSACTION
+            )
+        }
+
+        override val progressTracker = tracker()
+
+        /**
+         * The flow logic is encapsulated within the call() method.
+         */
+        @Suspendable
+        override fun call(): SignedTransaction {
+            progressTracker.currentStep = CREATING_COMMAND
+            //val signedTransaction = serviceHub.validatedTransactions.getTransaction(loanTxId)
+            // val signedTransaction = transactions.find { it.id == transactionHash } ?: throw IllegalArgumentException("Unknown transaction hash.")
+
+            val iouStateAndRef = serviceHub.iouStateFinder(UUID.fromString(iouStateLinearId))
+            val iouState = iouStateAndRef.state.data
+            val cashStatesAndRefs = serviceHub.moneyFinder(iouState.borrower, iouState.value)
```

Comment:
Your utility tool should also need the `creator: Party` of the money you wish to pay with.